python/lib/communication/dmod/communication/dataset_management_message.py

```diff
@@ -283,171 +253,62 @@ def __init__(self, action: ManagementAction, dataset_name: Optional[str] = None,
-
-    @property
-    def dataset_name(self) -> Optional[str]:
-        """
-        The name of the involved dataset, if applicable.
-
-        Returns
-        -------
-        Optional
-            The name of the involved dataset, if applicable.
-        """
-        return self._dataset_name
-
-    @property
-    def is_read_only_dataset(self) -> bool:
-        """
-        Whether the dataset involved is, should be, or must be (depending on action) read-only.
-
-        Returns
-        -------
-        bool
-            Whether the dataset involved is, should be, or must be (depending on action) read-only.
-        """
-        return self._is_read_only_dataset
-
-    @property
-    def management_action(self) -> ManagementAction:
-        """
-        The type of ::class:`ManagementAction` this message embodies or requests.
-
-        Returns
-        -------
-        ManagementAction
-            The type of ::class:`ManagementAction` this message embodies or requests.
-        """
-        return self._action
-
-    @property
-    def query(self) -> Optional[DatasetQuery]:
-        return self._query
-
-    def to_dict(self) -> Dict[str, Union[str, Number, dict, list]]:
-        serial = {self._SERIAL_KEY_ACTION: self.management_action.name,
-                  self._SERIAL_KEY_IS_READ_ONLY: self.is_read_only_dataset,
-                  self._SERIAL_KEY_IS_PENDING_DATA: self.is_pending_data}
-        if self.dataset_name is not None:
-            serial[self._SERIAL_KEY_DATASET_NAME] = self.dataset_name
-        if self.data_category is not None:
-            serial[self._SERIAL_KEY_CATEGORY] = self.data_category.name
-        if self.data_location is not None:
-            serial[self._SERIAL_KEY_DATA_LOCATION] = self.data_location
-        if self.data_domain is not None:
-            serial[self._SERIAL_KEY_DATA_DOMAIN] = self.data_domain.to_dict()
-        if self.query is not None:
-            serial[self._SERIAL_KEY_QUERY] = self.query.to_dict()
-        return serial
+        super().__init__(
+            management_action=action or data.pop("management_action", None),
+            dataset_name=dataset_name,
+            is_read_only_dataset=is_read_only_dataset or data.pop("read_only", False),
+            data_category=category or data.pop("data_category", None),
+            data_domain=domain or data.pop("data_domain", None),
+            data_location=data_location,
+            is_pending_data=is_pending_data or data.pop("pending_data", False),
+            query=query,
+            **data
+        )
+
+
+class DatasetManagementResponseBody(Serializable):
+    action: Optional[ManagementAction]
+    data_id: Optional[str]
+    dataset_name: Optional[str]
+    datasets: Optional[List[str]]
+    item_name: Optional[str]
+    # TODO: in the future, tighten the type restrictions of this field
+    query_results: Optional[Dict[str, Any]]
+    is_awaiting: bool = False
 
 
 class DatasetManagementResponse(Response):
```

Comment:
Not a dumb question. The parent class of `DatasetManagementResponse`, `Response`, includes a hierarchal field `data` that prior to the `pydantic` work was on `Optional[dict]`. One goal of the `pydantic` refactor was to improve the specificity of fields and as such, most if all of the `Response` subtypes' `data` fields were pulled out into `SomeSubtypeResponseBody`. This is mainly because `pydantic` does not offer a way to create hierarchy without embedding models within models. Meaning, I could not create the `DatasetManagementResponseBody` within the `DatasetManagementResponse` class. They had to be separate classes. Now that I think about it, im not sure that they _have to be_ separate entities. I think it is valid to do something like:


```python

class A(Serializable):
  class B(Serializable):
    field: int
  
  b: B 
```

It just looks and feels kind of wrong. Im not sure of its validity.

Reply:
So, it does work, it still just looks and feels non-pythonic. I can definitely be persuaded though.

```python
def test_it():
    m = A(b=A.B(field=12))
    print(m)
```

Thoughts?

Reply:
I was thinking more along the lines of running through (or creating a ticket to run through) the response classes and dropping the `data: ResponseBody` field for just adding what _would_ be in the `ResponseBody` class directly into the response. For example, instead of:

```python
class DatasetManagementResponse(Response):
    response_to_type: ClassVar[Type[AbstractInitRequest]] = DatasetManagementMessage

    data: DatasetManagementResponseBody
```

you'd have:

```python
class DatasetManagementResponse(Response):
    response_to_type: ClassVar[Type[AbstractInitRequest]] = DatasetManagementMessage

    action: Optional[ManagementAction]
    data_id: Optional[str]
    dataset_name: Optional[str]
    datasets: Optional[List[str]]
    item_name: Optional[str]
    # TODO: in the future, tighten the type restrictions of this field
    query_results: Optional[Dict[str, Any]]
    is_awaiting: bool = False
```

Having played around more with pydantic, I understand a little more of the approach. Go ahead and resolve the conversation if you don't have further responses.

Reply:
Yeah, I am not opposed to flattening these types out like you did above, however I have a feeling that we will want to do more than just flatten this type out in the future. This change is a little out of scope for this PR and requires more discussion. I have a feeling this and similar types will be more closely inspected as we move towards an event architecture.

Reply:
I've opened #347 to track investigating and potentially implementing these kind of changes.